src/libraries/System.Linq/src/System/Linq/AppendPrepend.SpeedOpt.cs

```diff
@@ -73,6 +73,13 @@ public override TSource[] ToArray()
             public override List<TSource> ToList()
             {
                 int count = GetCount(onlyIfCheap: true);
+
+                if (count == 1)
+                {
+                    // If GetCount returns 1, then _source is empty and only _item should be returned
+                    return new List<TSource>(1) { _item };
```

Comment:
I can see how this is preferable to doing `AddRange` on a singleton source, but how common a scenario is this to justify another branch? Couldn't it just be removed altogether?

Reply:
It certainly can be removed. I don't have any hard data on usage, but I do know that it's the sort of thing I feel like I encounter semi-frequently in my development. Cases like this example:

```c#
var result = source is not null
    ? source.Where(p => ...)
    : Enumerable.Empty<ItemType>();

if (otherItemToAppend is not null)
{
    result = result.Append(otherItemToAppend);
}

return result;
```

Of course, the example above is overly simplified and could be written to be more efficient with more branches. But this feels like the type of "this code is simpler and easier to read" approach that is commonly used.

That said, the most common case of appending a single item to an empty list would probably be a list that has been filtered using an `OfType` or `Where` predicate, which wouldn't trigger this fast path since `GetCount(onlyIfCheap: true)` will return `-1`.

Reply:
Ok, let's keep it for now. I likewise have no indication that the added branching could have real perf impact on non-singletons.